src/commonMain/kotlin/org/kbson/BsonBinary.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kbson
+
+/**
+ * A representation of the BSON Binary type.
+ *
+ * <p>Note that for performance reasons instances of this class are not immutable, so care should be
+ * taken to only modify the underlying byte array if you know what you're doing, or else make a
+ * defensive copy. </p>
+ */
+class BsonBinary(val type: Byte, val data: ByteArray) : BsonValue() {
```

Comment:
Shouldn't the type here be `BsonBinarySubType` instead? At least it would be more type safe.

Reply:
In Java its stored just as the type is stored as a byte - this might allow support for newer subtypes. 

I will ensure there is a BsonBinarySubType & ByteArray constructor.